WildAidDemo/functions/getDashboardStats/source.js

```diff
@@ -0,0 +1,169 @@
+        switch (key){
+          case "inspection.summary.safetyLevel":
+          aggregateTerms.$search.compound.must.push({
+            "search": {
+              "query": filter["inspection.summary.safetyLevel"],
+              "path":"inspection.summary.safetyLevel.level"
+            }
+          });
+          break;
+          case "date":
+          aggregateTerms.$search.compound.must.push({
+            "range": {
+              "path": "date",
+              "gte":  new Date(parseInt((new Date(filter.date)).valueOf()/86400000)*86400000),
+              "lte":  new Date((parseInt((new Date(filter.date)).valueOf()/86400000) + 1)*86400000-6400000)
+            }
+          });
+          break;
+          case "date-from":
+          aggregateTerms.$search.compound.must.push({
+            "range": {
+              "path": "date",
+              "gte":  new Date(parseInt((new Date(filter["date-from"])).valueOf()/86400000)*86400000)
+            }
+          });
+          break;
+          case "date-to":
+          aggregateTerms.$search.compound.must.push({
+            "range": {
+              "path": "date",
+              "lte":  new Date((parseInt((new Date(filter["date-to"])).valueOf()/86400000) + 1)*86400000-6400000)
+            }
+          });
+          break;
+          default:
+          aggregateTerms.$search.compound.must.push({
+            "search": {
+              "query": filter[key],
+              "path": key
+            }
+          })
+          break;
+        }
+      })
+    } else {
+      aggregateTerms = {
+        '$search': {
+          'term': {
+            'query': query,
+            'path': [
+              'vessel.name', 'captain.name', 'inspection.summary.violations.offence.explanation', 'inspection.summary.violations.offence.code', 'vessel.permitNumber', 'crew.license'
+            ],
+            'fuzzy': {
+              'maxEdits': 1.0
+            }
+          },
+          'highlight': {
+            'path': [
+              'vessel.name', 'inspection.summary.violations.offence.explanation', 'inspection.summary.violations.offence.code'
+            ]
+          }
+        }
+      }
+    }
+
+    aggregates.push(aggregateTerms);
+
+  }
+
+  let amount = boardingsCollection
+      .aggregate(aggregates.concat(
+        {
+          $count: "total"
+        }
+      )).next().then(r=> r ? r.total : 0);
+  let violations = boardingsCollection
+    .aggregate(aggregates.concat([{"$project": {
+          "violations": "$inspection.summary.violations",
+        }},
+        { $unwind: "$violations" },
+        { $count: "total" }
+    ])).next().then(r=> r ? r.total : 0);
+ /* let warnings = boardingsCollection
+    .aggregate(aggregates.concat([{"$project": {
+          "violations": "$inspection.summary.violations",
+        }},
+        { $unwind: "$violations" },
+        { $match: { "violations.disposition": "Warning" }},
+        { $count: "total" }
+    ])).next(); */
```

Comment:
Is there a reason to leave these commented-out lines in?

Reply:
Yep, we could delete it.